### ISSUE TITLE
Forge/Java incompatability

### ISSUE THREAD

**Issue by tehbasshunter** [simibubi/Above-and-Beyond#434]
On January 18th Oracle released an update for Java 8 as Java SE 8 Update 321. This unfortunately cause incompatibility with Forge 36.2.23 and earlier (They released 36.2.25 later but this still carried the incompatibility issue).  
  
This was solved by changing the used java version to Update 311 until Forge 36.2.26 was released.

At the moment 36.2.20 is currently shipped which causes the following error for users that import the modpack via MultiMC. This can be solved by changing Java version to the aforementioned version above or using Update 321 or newer along with Forge 36.2.26+ (Which would be ideal to have the modpack updated to).  

```
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: Caused by: java.lang.NoSuchMethodError: 'void sun.security.util.ManifestEntryVerifier.<init>(java.util.jar.Manifest)'
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.SecureJarHandler.createCodeSource(SecureJarHandler.java:66)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:275)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:136)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:98)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at java.base/java.lang.Class.forName0(Native Method)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at java.base/java.lang.Class.forName(Class.java:467)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at org.spongepowered.asm.service.modlauncher.ModLauncherClassProvider.findClass(ModLauncherClassProvider.java:67)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at org.spongepowered.asm.launch.platform.MixinConnectorManager.loadConnectors(MixinConnectorManager.java:70)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at org.spongepowered.asm.launch.platform.MixinConnectorManager.inject(MixinConnectorManager.java:59)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at org.spongepowered.asm.launch.platform.MixinPlatformManager.inject(MixinPlatformManager.java:196)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at org.spongepowered.asm.launch.MixinBootstrap.inject(MixinBootstrap.java:202)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at org.spongepowered.asm.launch.MixinLaunchPluginLegacy.initializeLaunch(MixinLaunchPluginLegacy.java:201)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at org.spongepowered.asm.launch.MixinLaunchPluginLegacy.initializeLaunch(MixinLaunchPluginLegacy.java:195)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.LaunchPluginHandler.lambda$announceLaunch$9(LaunchPluginHandler.java:97)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at java.base/java.util.HashMap.forEach(HashMap.java:1421)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.LaunchPluginHandler.announceLaunch(LaunchPluginHandler.java:97)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:52)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66)
[19:39:42] [main/INFO] [STDERR/]: [java.lang.Throwable:printStackTrace:659]: 	... 13 more
```

**Comment by fell-lucas** [comment]
Also had to manually update the forge installer to make the server work. Would be nice to update the curse forge releases.

**Comment by NugloFug** [comment]
having an issue trying to run the server I thought I set everything up correct but my server won't start
[latest.log](https://github.com/simibubi/Above-and-Beyond/files/8713090/latest.log)
[debug.log](https://github.com/simibubi/Above-and-Beyond/files/8713091/debug.log)
This is what I used to for the bat file: java -Xmx8G -Xms8G -Dsun.rmi.dgc.server.gcInterval=2147483646 -XX:+UnlockExperimentalVMOptions -XX:G1NewSizePercent=0 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M -XX:+UseG1GC -jar forge-1.16.5-36.2.34.jar nogui


 using java version 8 333 and forge version 1.16-36.2.34

**Comment by AbiDrew** [comment]
@NugloFug 
According to your debug.log you are using java version 17.0.1. Which is unsupported. You mentioned a bat file, which would mean you are trying to run this on a Windows based server. I have not personally ever tried to run a modded Minecraft server on a Windows based server, so I can't really offer you much in the way of troubleshooting that problem, but your bat command line appears to be calling system installed java, which is likely java 17. If you have a different java version installed somewhere else you'll need to specify that somehow. You'll need to look up how to handle multiple installed versions of java on a Windows server yourself though, that's not related to the issue that's reported here and we don't really provide support for budding server owners in things not related to the pack.

@tehbasshunter @fell-lucas 
You have already figured out you can just install the latest recommended forge server yourselves to work around this issue for the time being. We are working on a future release right now, which has an indefinite time frame. We will likely be using whatever the latest recommended forge is at the time of that release which will close this issue entirely at that time. Unless that breaks something, in which case we would use the last working recommended release. For now, folks are just going to have to continue to update forge on their own.

**Comment by NugloFug** [comment]
@AbiDrew sorry first time using github do you know where I could look for help this is the only pack that has given me issues in starting a server for.

**Comment by AbiDrew** [comment]
Just use Google or your favorite search engine. If it was me, I'd start with "multiple versions java windows how" and based on the results of that query, if nothing quite works, but some of it at least offers more clues on what to look for, I'd correct my query and keep drilling down to the solution.

**Comment by fell-lucas** [comment]
@AbiDrew that's cool, but the issue is affecting both client and server .zip files, normal people who just want to download and play shouldn't have to google stuff themselves to fix something that isn't working. It's a trivial task to reupload at least the most recent release v1.3 with the latest Forge .jar.

In my experience, there was a lot of friction to downloading and running correctly the server pack. I've never had to run a Forge installer on any modpack server before, nor does the wiki say you have to pass --installServer to it. By making it as accessible and 'plug and play' as possible to normal people, maybe through a `start.sh` script like most of the modded servers do, you'd increase the overall satisfaction and download count for the modpack. 

Not everyone is used to googling stuff as much as developers. Some people just want to play.

**Comment by AbiDrew** [comment]
Interesting that anyone is having issues single-player or client-side at all. If they are, they're using a third-party launcher, not the official, and the only third-party launcher that I know for sure is being tested is GD Launcher, because that's the one I'm using. Both Curseforge and GDL use their own java rather than system java. I'm not sure why simibubi has not released 1.3 with only the forge version changed, calling it 1.3.1 to avoid any issues from same version numbering. I'm here for the big update. @simibubi apparently this issue is not just affecting servers but also clients with third-party launchers. Might be grounds for an emergency patch version?

**Comment by AbiDrew** [comment]
Also, Lucas, I'd be happy to add stuff to the next big update to make server install and management a lot easier... That's going to be a bit more long-term though. I wanted to make an "install.sh" for my custom systemd service units anyways, and this makes a pretty good place for testing that.

**Comment by Diagnul** [comment]
I ran into this same issue today while trying to start a fresh server for Create A&B 1.3 using the latest GA release of OpenJDK (11.0.15 2022-04-19) on Debian 11.

`java.lang.NoSuchMethodError: 'void sun.security.util.ManifestEntryVerifier.<init>(java.util.jar.Manifest)'
`

I was able to work around the issue using the information provided by tehbasshunter. I downloaded the Forge 36.2.26 installer and installed that instead of Forge 36.2.20 which comes with the Create A&B 1.3 server files.

With that version of forge the server launched correctly, and I was also able to connect to the server with a client that was still using Forge 36.2.20
